### ISSUE TITLE
cross-source data contamination

### ISSUE THREAD

**Issue by nyurik** [vega/vega#1239]
This graph shows the same number of rows (7) in each data source, even though `nest` transform should only affect `src2`. Removing all src1 transforms makes them different. 

If someone could show me a similar unit test, I would submit this (and other bugs) as a failing unit tests. Thx!

```json
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "autosize": "pad",
  "padding": 5,
  "width": 500,
  "height": 500,
  "title": {
    "text": {
      "signal": "'src1:'+length(data('src1')) + ', src2:'+length(data('src2')) + ', src3:'+length(data('src3'))"
    }
  },
  "data": [
    {
      "name": "src1",
      "values": {
        "aggregations": {
          "country": {
            "buckets": [
              {
                "key": "391",
                "customer": {
                  "buckets": [
                    {
                      "key": "abc",
                      "doc_count": 100,
                      "disputed": {
                        "buckets": [
                          {"key": "Yes", "doc_count": 50},
                          {"key": "No", "doc_count": 30}
                        ]
                      }
                    }
                  ]
                }
              }
            ]
          }
        }
      },
      "format": {"property": "aggregations.country.buckets", "type": "json"},
      "transform": [
        {"type": "flatten", "fields": ["customer.buckets"], "as": ["val1"]},
        {"type": "flatten", "fields": ["val1.disputed.buckets"], "as": ["val2"]},
        {"type": "formula", "expr": "datum.val1.key", "as": "val"},
        {"type": "formula", "expr": "datum.val2.key", "as": "val2"}
      ]
    },
    {
      "name": "src2",
      "source": "src1",
      "transform": [
        {"type": "nest", "keys": ["val0", "val1", "val2"], "generate": true}
      ]
    },
    {"name": "src3", "source": "src1"}
  ]
}
```

**Comment by jheer** [comment]
Thanks! This is indeed a bug, caused by improper pulse handling in the nest transform. I have a fix and will include it in the next release.

**Comment by jheer** [comment]
Fixed in v3.3.0!
